jwst/ramp_fitting/tests/test_ramp_fit_cases.py

```diff
@@ -0,0 +1,541 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+
+from jwst.ramp_fitting.ramp_fit import ramp_fit
+from jwst.datamodels import MIRIRampModel
+from jwst.datamodels import GainModel, ReadnoiseModel
+
+#
+# Each of these 12 tests is for a single ramp in a single integration. The ramps
+#  have a variety of GROUPDQ vectors, with 1 or more segments in each ramp.  The
+#  comparison of the PRIMARY output results are partly to verify the slopes and
+#  variances of the combination of the segments in a ramp within the single
+#  integration.  The comparison of the OPTIONAL output results are to verify the
+#  results for each of the individual segments in a ramp.  Within each test is a
+#  description of classification ('CASE') within the code of all of the segments
+#  for the pixel based on the ramp's GROUPDQ, and the resulting segments and
+#  their SCI values (these are mostly for my reference).
+#
+
+def test_pix_0():
+    """
+    CASE A: segment has >2 groups, at end of ramp.
+    SCI seg is [15., 25., 35., 54., 55., 65., 75., 94., 95., 105.](A)
```

Comment:
It may not matter to the actual test, but it appears that you're intending to simulate a ramp that has accumulating signal of 10 DN per group, with groups 4 and 8 having anomalous values relative to that steady increase of 10 DN per group. If groups 4 and 8 are supposed to represent something like CR hits (they have 19 DN of signal, instead of the normal 10, for an excess of 9), all of the groups following those simulated hits should continue to carry along the extra 9 DN of signal from the hits, i.e. the sequence would be [15, 25, 35, 54, 64, 74, 84, 103, 113, 123], so that the last 2 groups have a total excess signal of 18 DN relative to normal. Or are CR hits in groups 4 and 8 not what you're intending to simulate here?